### ISSUE TITLE
Incorrect ipv6 format in istio-proxy

### ISSUE THREAD

**Issue by dkushwaha07** [istio/istio#37079]
### Bug Description

In 1.10, issue observed in istio-proxy health checking mechanism.  Istio trying to use one of the eth0 host ip to handling the traffic, but it doesn’t contain the bracket [] for http host header and it cause issue to reply . 

### Version

```prose
[root@ms001 ~]# istioctl version
client version: 1.10.4
control plane version: 1.10.4
data plane version: 1.10.4 (4 proxies)
 
[root@ms001 ~]# kubectl version --short
Client Version: v1.21.5
Server Version: v1.21.5
```

### Affected product area

- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [X] Developer Infrastructure
- [ ] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

### Is this the right place to submit this?

- [X] This is not a security vulnerability
- [X] This is not a question about how to use Istio

**Comment by howardjohn** [comment]
There have been a lot of issues just like this fixed since 1.10. I strongly suspect this is fixed (as we now also have better test coverage). Please re-open if its not in the `master` branch.

Note that ipv6 support is 'alpha' and not production ready, so there are still issues being worked out.

Cheers

**Comment by dkushwaha07** [comment]
@howardjohn we didn't face this issue in 1.9, but getting it in 1.10
